tests/unit/acquisition/test_utils.py

```diff
@@ -64,3 +71,25 @@ def test_select_nth_output() -> None:
 
     assert np.all(select_nth_output(a) == a[..., 0])
     assert np.all(select_nth_output(a, 3) == a[..., 3])
+
+
+@pytest.mark.parametrize(
+    "x, other_x",
+    [
+        (tf.constant([[[1], [2]]]), tf.constant([[1], [2]])),
+        (tf.constant([[1], [2]]), tf.constant([[[1], [2]]])),
+        (tf.constant([[1, 2], [2, 3]]), tf.constant([[1], [2]])),
+    ],
+)
+def test_random_mixing_raises_for_invalid_input(x: TensorType, other_x: TensorType) -> None:
+    with pytest.raises(ValueError):
+        randomly_mix_x_with_other_x(x, other_x, 0.5)
+
+
+@random_seed
+def test_random_mixing_raises_works() -> None:
+    x = tf.random.normal([10_000, 6])
+    x_other = tf.random.normal([1, 6])
+    x_mixed = randomly_mix_x_with_other_x(x, x_other, 0.5)
```

Comment:
Could you please also parametrize by:

1. prob? (i.e. don't just test 0.5)
2. tf.function compilation? (like in test_randomize_hyperparameters_randomizes_kernel_parameters_with_priors and various other tests)